### ISSUE TITLE
Feature: Improve error message reported by getParserServices

### ISSUE THREAD

**Issue by disaacson** [typescript-eslint/typescript-eslint#8482]
### Suggestion

I'm getting an error from this line
https://github.com/typescript-eslint/typescript-eslint/blame/18c3216485c381dd0215cfa7099f33bc721c8efe/packages/utils/src/eslint-utils/getParserServices.ts#L93

```
Error: Error while loading rule '@typescript-eslint/no-base-to-string': You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser.
Note: detected a parser other than @typescript-eslint/parser. Make sure the parser is configured to forward "parserOptions.project" to @typescript-eslint/parser.
```

I would like to submit a pull request to include in the error message the `parserPath` value that is causing the error. That could help with debugging this issue when it arises.

**Comment by bradzacher** [comment]
Could you please provide some more info here - what usecase are you looking to improve by adding the parser path?

I personally am not convinced that it's a good thing based on the usecases I know of and would think the extra information would be noise in those cases - so I'm going to need some more information from you.

**Comment by disaacson** [comment]
I've had to debug this issue and it is difficult in a large monorepo project without knowing that value. As the error is about, or based on, that specific value, it would be very helpful information in debugging that error when it occurs.

**Comment by bradzacher** [comment]
> As the error is about, or based on, that specific value

Sorry that's the part I don't understand.
How is the error about the parser path?

How would knowing that the parser path is `node_modules/@typescript-eslint/dist/index.js` help?

Could you please describe how knowing that would have helped you debug?

**Comment by disaacson** [comment]
Here is the output when linting the project:
```
Error: Error while loading rule '@typescript-eslint/no-base-to-string': You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser.
Note: detected a parser other than @typescript-eslint/parser. Make sure the parser is configured to forward Occurred while linting /Users/derrick/code/borderless-web/packages/utils-shared-configs/package.json
   at throwError (/Users/derrick/code/borderless-web/packages/utils-shared-configs/node_modules/@typescript-eslint/utils/dist/eslint-utils/getParserServices.js:39:11)
   at getParserServices (/Users/derrick/code/borderless-web/packages/utils-shared-configs/node_modules/@typescript-eslint/utils/dist/eslint-utils/getParserServices.js:23:9)
   at create (/Users/derrick/code/borderless-web/packages/utils-shared-configs/node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-base-to-string.js:69:55)
   at Object.create (/Users/derrick/code/borderless-web/packages/utils-shared-configs/node_modules/@typescript-eslint/utils/dist/eslint-utils/RuleCreator.js:38:20)
   at createRuleListeners (/Users/derrick/code/borderless-web/node_modules/eslint/lib/linter/linter.js:895:21)
   at /Users/derrick/code/borderless-web/node_modules/eslint/lib/linter/linter.js:1066:110
   at Array.forEach (<anonymous>)
   at runRules (/Users/derrick/code/borderless-web/node_modules/eslint/lib/linter/linter.js:1003:34)
   at Linter._verifyWithoutProcessors (/Users/derrick/code/borderless-web/node_modules/eslint/lib/linter/linter.js:1355:31)
   at Linter._verifyWithoutProcessors (/Users/derrick/code/borderless-web/node_modules/eslint-plugin-eslint-comments/lib/utils/patch.js:166:36)
error Command failed with exit code 2.
```

It says `detected a parser other than @typescript-eslint/parser`, but it does not say which one. The project is configured to use `@typescript-eslint/parser`, so it is confusing. If the error said what this `other parser` is that is the cause of the error, it could help the developer identify which parser is being found in place of the supposedly configured one.

In short, it refers to this "other" parser as being the cause of the error. It would be very helpful to know which one is unintentially being found. That would aid in tracking down the root cause to the error.

**Comment by disaacson** [comment]
Using the example in the automated test for that error, it would now output
```
You have used a rule which requires parserServices to be generated. You must therefore provide a value for the \"parserOptions.project\" property for @typescript-eslint/parser.
    Parser: @babel/parser.js
    Note: detected a parser other than @typescript-eslint/parser. Make sure the parser is configured to forward \"parserOptions.project\" to @typescript-eslint/parser.
```
The information that the parser it found was @babel/parser.js can help track down the cause of the error.

**Comment by JoshuaKGoldberg** [comment]
Seems reasonable to me to mention the parser explicitly. The downside is having a bit more verbose of a log, but the upside of being more explicit & clear can IMO help folks figure out where things are going wrong. So, what disaacson said. 😄 

cc @bradzacher - thoughts?

**Comment by bradzacher** [comment]
sure sounrds good

**Comment by disaacson** [comment]
I updated the PR's checklist. Let me know if you have any feedback or requested changes. Thanks!
https://github.com/typescript-eslint/typescript-eslint/pull/8484/files
